website/scripts/ros/ros_system_alert.js

```diff
@@ -10,104 +10,119 @@
 function checkSystemAlerts() {
 
     // Subscribing to a Topic
-    listenerSystemAlert = new ROSLIB.Topic({
-        ros: g_ros,
-        name: T_SYSTEM_ALERT,
-        messageType: M_SYSTEM_ALERT
-    });
-    let systemAlertQueue = [];
-    // Then we add a callback to be called every time a message is published on this topic.
-    listenerSystemAlert.subscribe(function (message) 
+    // Check if our global subscription variable is defined already
+    if (typeof listenerSystemAlert === 'undefined' || listenerSystemAlert === null) 
     {
-         //Check ROSBridge connection before subscribe a topic
-         IsROSBridgeConnected();
-        var messageTypeFullDescription = 'NA';
-        console.log("system alert type" + message.type);
-        switch (message.type) 
+        // Initialize a topic object
+        listenerSystemAlert = new ROSLIB.Topic({
```

Comment:
type check after definition?

Reply:
Not quite following what you are asking here. 
What are you asking to type check?